server/internal/mixkey/mixkey.go

```diff
@@ -136,138 +106,20 @@ func (k *MixKey) IsReplay(rawTag []byte) bool {
 	var tag [TagLength]byte
 	copy(tag[:], rawTag)
 
-	// Check the bloom filter for the tag, to see if it might be a replay.
-	maybeReplay, inWriteBack := k.testAndSetTagMemory(&tag)
-	if !maybeReplay {
-		// k.isNotReplay() will add the tag to the write-back cache, so
-		// just poke the flush routine and return.
-		select {
-		case k.flushCh <- true:
-		default:
-			// Non-blocking channel write, because the channel is buffered
-			// and has a timer fallback.
-		}
-		return false
-	}
-
-	// Slow path, either a false positive or a replay.
-	//
-	// Note: It alternatively would be acceptable to just drop the packet,
-	// but k.isNotReplay()'s behavior will need to change on filter
-	// saturation.
-
-	isReplay := inWriteBack
-	if !isReplay {
-		// Well, it's not in the write-back cache, so query the database.
-		//
-		// Since we're stuck hitting the database anyway, might as well
-		// bypass the cache and save ourselves some pain by doing the
-		// insertion here.
-		if err := k.db.Update(func(tx *bolt.Tx) error {
-			bkt := tx.Bucket([]byte(replayBucket))
-			isReplay = testAndSetTagDB(bkt, tag[:])
-			return nil
-		}); err != nil {
-			panic("BUG: mixkey: Failed to query the replay filter: " + err.Error())
-		}
-	}
-	return isReplay
-}
-
-func testAndSetTagDB(bkt *bolt.Bucket, tag []byte) bool {
-	// Retrieve the counter from the database for the tag if it exists.
-	//
-	// XXX: The counter isn't actually used for anything since it isn't
-	// returned.  Not sure if it makes sense to keep it, but I don't think
-	// it costs us anything substantial to do so.
-	var seenCount uint64
-	if b := bkt.Get(tag); b != nil {
-		if len(b) == 8 {
-			seenCount = binary.LittleEndian.Uint64(b)
-		} else {
-			// Treat invalid but present entries as being seen.
-			seenCount = 1
-		}
-	}
-	seenCount++         // Increment the counter by 1.
-	if seenCount == 0 { // Should never happen ever, but handle correctly.
-		seenCount = math.MaxUint64
-	}
-
-	// Write the (potentially incremented) counter.
-	var seenBytes [8]byte
-	binary.LittleEndian.PutUint64(seenBytes[:], seenCount)
-	if err := bkt.Put(tag, seenBytes[:]); err != nil {
-		panic(err)
-	}
-	return seenCount != 1
-}
-
-func (k *MixKey) testAndSetTagMemory(tag *[TagLength]byte) (bool, bool) {
 	k.Lock()
 	defer k.Unlock()
 
-	// TODO/perf: Perhaps the lock should only cover the bloom filter,
-	// and a sync.Map used for the pending write-back entries.
-
-	// If the filter is saturated then force a database lookup.
+	// If the filter is saturated then probability of a false replay is increased
+	// XXX: the filter size should be turned for the maximum line rate expected so that this does not happen
 	if k.f.Entries() >= k.f.MaxEntries() {
-		return true, k.writeBack[*tag]
+		// XXX: does not have a logger...
```

Comment:
1. remove the comment about not having a logger or 2. add the logger. pick one. you must choose between those two options.